feat: Form Builder

shariquerik opened the pull request:
### Started fresh PR with vue 3 https://github.com/frappe/frappe/pull/18381

This feature aims to help the user edit the DocType's fields layout directly from UI (Form Builder), rather than from the form view of the DocType.

Triggered by https://github.com/frappe/frappe/pull/17785

**Key Features:**
- [x] Drag & Drop fields from the sidebar to form
- [ ] Add New Tabs, Sections & Columns
- [ ] Delete Tabs, Sections, Columns & Fields
- [x] Edit the details of Tabs, Sections & Columns
- [x] Drag & Drop to move around fields in the form and tabs
- [ ] Group fields and move group of fields
- [ ] Save and update the changes
- [ ] Should also work for Customize Form fields to add and update custom fields


**Make public & stable:**
- [ ] UI Test
- [ ] Documentation

>no-docs

shariquerik committed:
feat: Form Builder
- Form Builder Page
- Vue setup
- Render doctype fields in tabs, sections, column
- Drag & Drop implementation

shariquerik committed:
fix: added button to try new form builder in doctype

also added button to go to doctype from form builder

shariquerik committed:
fix: added all_fieldtype list in frappe.model

shariquerik committed:
feat: FormControls Section to add fields to the form

ankush commented:
review checklist: 
- [x] revert parts of this which aren't required anymore https://github.com/frappe/frappe/pull/17785

shariquerik committed:
feat: form field detail section

- Render field details in sidebar section, two way binding
- hide docfield based on depends_on expression
- common search for fields and fields detail

shariquerik committed:
revert: form editor code

shariquerik committed:
Merge branch 'develop' into form-builder

shariquerik commented:
Since the code was built on vue 2 opening another PR https://github.com/frappe/frappe/pull/18381 for Form Builder with vue 3
